### PR TITLE
🪟 🎨 Trial end banner

### PR DESCRIPTION
**Pull request by teallarson**
## What
Closes https://github.com/airbytehq/airbyte-cloud/issues/1488

Adds a trial end banner.

Screen Shot 2022-07-21 at 1 03 09 PM

## How
Cloud has two possible banners -- trial end and out of credits.  The `MainView` component  for Cloud now considers which banner to show and calculates its contents, passing it to our generic `AlertBanner` component (which has been converted to `.scss`).

## Recommended reading order
1. `MainView`
2. `AlertBanner`
3. others

### CODE REVIEW

File: airbyte-webapp/src/packages/cloud/services/workspaces/WorkspacesService.tsx [hunk 1]
```diff
@@ -108,7 +108,11 @@ export function useRemoveWorkspace() {
 export function useGetCloudWorkspace(workspaceId: string): CloudWorkspace {
   const service = useGetWorkspaceService();
 
-  return useSuspenseQuery<CloudWorkspace>([workspaceKeys.detail(workspaceId)], () => service.get(workspaceId));
+  const workspace = useSuspenseQuery<CloudWorkspace>([workspaceKeys.detail(workspaceId)], () =>
+    service.get(workspaceId)
+  );
+  workspace.trialExpiryTimestamp = 1658577284;
```
Comment (edmundito) [inline airbyte-webapp/src/packages/cloud/services/workspaces/WorkspacesService.tsx:114]: Was this change intended or just for testing?

File: airbyte-webapp/src/packages/cloud/views/layout/MainView/MainView.tsx [hunk 2]
```diff
@@ -36,25 +38,51 @@ const DataBlock = styled.div<{ hasBanner?: boolean }>`
...
-  const showBanner =
+  const showCreditsBanner =
     cloudWorkspace.creditStatus &&
     [
       CreditStatus.NEGATIVE_BEYOND_GRACE_PERIOD,
       CreditStatus.NEGATIVE_MAX_THRESHOLD,
       CreditStatus.NEGATIVE_WITHIN_GRACE_PERIOD,
-    ].includes(cloudWorkspace.creditStatus);
+    ].includes(cloudWorkspace.creditStatus) &&
+    !cloudWorkspace.trialExpiryTimestamp;
```
Comment (edmundito) [inline airbyte-webapp/src/packages/cloud/views/layout/MainView/MainView.tsx:51]: In this case, if the `trialExpiryTimestamp` is null-ish, then it means that it's not expired?

File: airbyte-webapp/src/packages/cloud/views/layout/MainView/MainView.tsx [hunk 3]
```diff
@@ -36,25 +38,51 @@ const DataBlock = styled.div<{ hasBanner?: boolean }>`
...
+
+  const alertToShow = showCreditsBanner ? "credits" : cloudWorkspace.trialExpiryTimestamp ? "trial" : null;
```
Comment (edmundito) [inline airbyte-webapp/src/packages/cloud/views/layout/MainView/MainView.tsx:53]: Nit: In typescript, should be `undefined` instead of null because `null` is considered a different type

File: airbyte-webapp/src/packages/cloud/views/layout/MainView/MainView.tsx [hunk 4]
```diff
@@ -36,25 +38,51 @@ const DataBlock = styled.div<{ hasBanner?: boolean }>`
...
+
+  const generateAlert = useMemo(() => {
```
Comment (edmundito) [inline airbyte-webapp/src/packages/cloud/views/layout/MainView/MainView.tsx:55]: `alertMessage`

**Review by edmundito — COMMENTED** [review]
Looks good. I had a question about a particular value in the service and some tiny suggestions

File: airbyte-webapp/src/packages/cloud/views/layout/MainView/MainView.tsx (hunk 2 above)
Comment (tealjulia) [reply airbyte-webapp/src/packages/cloud/views/layout/MainView/MainView.tsx:51]: If the `trialExpiryTimestamp` is null-ish then the user is not in a trial (that is how the API was configured by Cloud team).

**Review by timroes — APPROVED** [review]
Code LGTM, tested locally with a trial workspace and banner shows as expected.
